packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterInvalidStateException.kt

```diff
@@ -18,5 +18,5 @@ package com.amazonaws.amplify.amplify_auth_cognito.types
 import com.amplifyframework.auth.AuthException
 
 class FlutterInvalidStateException: AuthException {
-    constructor(message: String, recoverySuggestion: String) : super(message, Exception("invalidSate"), recoverySuggestion)
+    constructor(message: String, recoverySuggestion: String) : super(message, Exception("invalidState"), recoverySuggestion)
```

Comment:
🕵️   Is it possible that someone could be looking for `invalidSate` for some reason in their code?

Reply:
🤯  ... Yes, it is definitely possible.

I am not sure if we would consider the message attached to an error part of the public API or not. It depends how you define public API (anything that is technically exposed as public or only the intended use).

Either way, this IMO is a bug fix and bug fixes should not be considered breaking in the same way that changing working functionality is considered breaking. Bug fixes are fixes to something that is currently broken. This entire PR is actually a good example of that. Currently the wrong exception is being thrown. This PR changes which exception type is being thrown. That could conceivably break code that a customer has written, but it is a fix to something that is broken and was never intended to be used that way.